### DISCUSSION TITLE
Auger control

### DISCUSSION THREAD

**Discussion by jazzmonger** [jazzmonger/wood-pellet-stove-with-TYWE1S-Tuya-chip#14] [General]
Controlling the auger has been more complicated than I originally thought. This is a burning stove in my living room. Get it wrong and nothing good happens. There is a temp_protect sensor that is supposed to trigger and shut everything down if the stove gets too hot, but I don't want to rely on that.   

So, adding lots of checks to all sorts of conditions before turning on the alternate P5 power level has consumed many hours of fiddling.  Thankfully I have a spare MCU unit and display board on my test bench. This lets me test all kinds of wacky scenarios and I've now mapped ALL error messages to real text that is helpful. I need to update the sensors file in the files section.  [done]

one last piece I'm adding right now is a heartbeat sensor that makes sure the auger is turning when it should be, and is not dead, and restart the esp to fix it. That would cause the E1 error, fire went out. This is proving tricky.

I also have to test everything when home assistant is dead.  All the code runs local and internal to the esp but there is still an api connection to home assistant. I've managed to crash home assistant at several times doing who knows what, requiring a restart of both core and supervisor, so this testing step is important. If home assistant goes down, this chip should just plug along like nothing happened.

**Comment by jazzmonger** [comment]
Finally nailed it.  See the files section for the new config file.  pics of the mods to come.

**Reply by Deep-Six** [reply]
Hey @jazzmonger  firstly happy Christmas thanks so much for this work automating my pellet stove in HA is exactly what I'm looking for and I can't believe I'm staring at nearly a solution. My question is I have a England Stove Works 25, it has a control board, it looks like you're hacking some sort of "smart controller" are these interchangeable with the side panel control board, or am I out of luck for automating my system.

**Reply by Deep-Six** [reply]
![20221224_123247.jpg](https://user-images.githubusercontent.com/23326167/209453229-b91eb225-bca1-47fa-b083-b46a353460be.jpg)

**Comment by jazzmonger** [comment]
Depends on your controller. Send me a pic of what it looks like or a web link to the stove itself.

is it Tuya based? What does the app look like?

**Reply by Deep-Six** [reply]
See above :)

**Reply by jazzmonger** [reply]
Oh boy... that's an old one! I'm afraid everything I have here is for the stoves with a specific controller based on Tuya.

that said, if you're so inclined, you could likely adapt my controller code to control your stove, but it would be some work. And you'd have to build a triac based power interface that controls the speeds of the auger and blower motors.  I seem to recall seeing a YouTube video of a guy who adapted an old wood stove with a new Wifi controller.  Search YouTube!

**Comment by jazzmonger** [comment]
I'm now working on the loss of api problem now when home assistant goes dark.  I think there may be a bug in the ESPHome code. 

![20A079DD-D0D2-4398-9D51-C1219F1C68B6](https://user-images.githubusercontent.com/52110065/209446235-f3be7e56-b91c-4a05-9e80-941744cf22af.jpeg)
Hopefully the code jockeys on discord can help me solve this one.

Edit: all working correctly. It turns out that once the on_shutdown routine is entered, no more log entries are displayed, but all the code executes.
